### PR TITLE
Fix/merge conflict

### PR DESCRIPTION
**Pull request by Hakdoff**
OVERVIEW
Fixed merging issues and fixed routes.

### CODE REVIEW

File: src/app/Http/Controllers/AuthController.php [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+use Illuminate\Support\Facades\Session;
+
+class AuthController extends Controller
+{
+
+    public function register(RegisterRequest $request)
+    {
+        try {
+
+
```
Comment (hipe01mark) [inline src/app/Http/Controllers/AuthController.php]: trail spacing!

File: src/app/Http/Controllers/AuthController.php [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+use Illuminate\Support\Facades\Hash;
+use App\Http\Requests\RegisterRequest;
+use Illuminate\Auth\Events\Registered;
+use Illuminate\Support\Facades\Session;
+
+class AuthController extends Controller
+{
+
+    public function register(RegisterRequest $request)
+    {
```
Comment (hipe01mark) [inline src/app/Http/Controllers/AuthController.php]: 1.) Add docbloc without parameters and return types.
2.) Add return types.

File: src/app/Http/Controllers/AuthController.php [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+            auth()->login($user);
+            event(new Registered($user));
+
+            return redirect('/verify-waiting')->with('success', 'Registration is successful! Please check your email for verification.');
+        } catch (\Exception $e) {
+            return redirect('/register-page')->with('error', 'Registration failed: ' . $e->getMessage());
+        }
+    }
+
+    public function verifyEmail($id, $hash)
```
Comment (hipe01mark) [inline src/app/Http/Controllers/AuthController.php]: 1.) Add docbloc without parameters and return types.
2.) Add return types.

File: src/app/Http/Controllers/AuthController.php [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+        $user->status = 1;
+        $user->save();
+
+        event(new Verified($user));
+
+        return redirect('/dashboard')->with('verified', true);
+    }
+
+    public function emailVerifyRedirect()
```
Comment (hipe01mark) [inline src/app/Http/Controllers/AuthController.php]: 1.) Add docbloc without parameters and return types.
2.) Add return types.

File: src/app/Http/Controllers/AuthController.php [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    {
+        return view('pages/auth/register');
+    }
+    public function resendEmailVerification()
```
Comment (hipe01mark) [inline src/app/Http/Controllers/AuthController.php:85]: 1.) Add docbloc without parameters and return types.
2.) Add return types.

File: src/app/Http/Controllers/AuthController.php [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    {
+        $user = auth()->user();
+        $user->sendEmailVerificationNotification();
+        return redirect()->back()->with('success', 'Email verification link resent.');
+    }
+
+    public function logout()
```
Comment (hipe01mark) [inline src/app/Http/Controllers/AuthController.php]: 1.) Add docbloc without parameters and return types.
2.) Add return types.

File: src/app/Http/Controllers/DashboardController.php [hunk 7]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+
+class DashboardController extends Controller
+{
+    public function dashboard()
```
Comment (hipe01mark) [inline src/app/Http/Controllers/DashboardController.php:9]: 1.) Add docbloc without parameters and return types.
2.) Add return types.

File: src/app/Http/Controllers/ProfileController.php [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+namespace App\Http\Controllers;
+
+use App\Models\Profile;
+use Illuminate\Http\Request;
+use App\Http\Requests\RegisterRequest;
+
+class ProfileController extends Controller
+{
+    public function createProfile($request, $userId)
```
Comment (hipe01mark) [inline src/app/Http/Controllers/ProfileController.php]: 1.) Add docbloc without parameters and return types.
2.) Add return types.

File: src/app/Http/Controllers/UserController.php [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+namespace App\Http\Controllers;
+
+use App\Models\User;
+use Illuminate\Support\Str;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Hash;
+
+class UserController extends Controller
+{
+    public function createUser($request)
```
Comment (hipe01mark) [inline src/app/Http/Controllers/UserController.php]: 1.) Add docbloc without parameters and return types.
2.) Add return types.

File: src/app/Models/Profile.php [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+    /**
+     * Get the attributes that should be cast.
+     *
+     * @return int
+     */
+    public function user()
+    {
+        return $this->belongsTo(User::class, 'user_id');
+    }
+}
```
Comment (hipe01mark) [inline src/app/Models/Profile.php]: Trail spacing!

File: src/app/Models/User.php [hunk 11]
```diff
@@ -44,4 +46,18 @@ protected function casts(): array
             'password' => 'hashed',
         ];
     }
+    /**
+     * Get the attributes that should be cast.
+     *
+     * @return int
```
Comment (hipe01mark) [inline src/app/Models/User.php]: remove parameters and return types in ever docbloc

File: src/app/Models/User.php [hunk 12]
```diff
@@ -44,4 +46,18 @@ protected function casts(): array
...
+     */
+    public function profile()
```
Comment (hipe01mark) [inline src/app/Models/User.php]: add return type

File: src/app/Models/User.php [hunk 13]
```diff
@@ -44,4 +46,18 @@ protected function casts(): array
...
+    {
+        return $this->hasOne(Profile::class, 'user_id');
+    }
+
+    public function sendEmailVerificationNotification()
```
Comment (hipe01mark) [inline src/app/Models/User.php]: 1.) Add docbloc without parameters and return types.
2.) Add return types.

File: src/app/Notifications/VerifyEmail.php [hunk 14]
```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace App\Notifications;
+
+use Illuminate\Auth\Notifications\VerifyEmail as VerifyEmailNotification;
+use Illuminate\Notifications\Messages\MailMessage;
+
+class VerifyEmail extends VerifyEmailNotification
+{
+    protected function buildMailMessage($url)
```
Comment (hipe01mark) [inline src/app/Notifications/VerifyEmail.php]: 1.) Add docbloc without parameters and return types.
2.) Add return & parameter types.

File: src/database/migrations/2024_04_04_073207_create_profiles_table.php [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+    }
+
+    /**
+     * Reverse the migrations.
+     */
+    public function down(): void
+    {
+        Schema::dropIfExists('profiles');
+    }
+};
```
Comment (hipe01mark) [inline src/database/migrations/2024_04_04_073207_create_profiles_table.php]: trail spacing!

If you see someting like this in your PR just add breakline

image

File: src/resources/css/app.css [hunk 16]
```diff
@@ -0,0 +1,3 @@
+@tailwind base;
+@tailwind components;
+@tailwind utilities;
```
Comment (hipe01mark) [inline src/resources/css/app.css]: trail spacing!

If you see someting like this in your PR just add breakline

File: src/resources/scss/app.scss [hunk 17]
```diff
@@ -0,0 +1 @@
+@use './auth/login';
```
Comment (hipe01mark) [inline src/resources/scss/app.scss]: trail spacing!

If you see someting like this in your PR just add breakline

File: src/resources/scss/auth/login.scss [hunk 18]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    &__padding4 {
+        @apply py-4;
+    }
+
+    &__padding5 {
+        @apply py-5;
+    }
+}
```
Comment (hipe01mark) [inline src/resources/scss/auth/login.scss]: trail spacing!

If you see someting like this in your PR just add breakline

File: src/resources/views/app.blade.php [hunk 19]
```diff
@@ -0,0 +1,9 @@
+@extends('pages.layouts.app')
+
+@section('content')
+
+<div class="login">
+    @extends('pages.auth.login')
+</div>
+
+@endsection
```
Comment (hipe01mark) [inline src/resources/views/app.blade.php]: trail spacing!

If you see someting like this in your PR just add breakline

File: src/resources/views/pages/auth/dummy-dashboard.blade.php [hunk 20]
```diff
@@ -0,0 +1,21 @@
...
+    @else
+    <p>test( 0 = unverified, 1 = verified) status: <b>{{auth()->user()->status}}</b></p>
+    @endif
+
+    <a href="/logout">
+        <button>logout</button>
+    </a>
+</body>
+
+</html>
```
Comment (hipe01mark) [inline src/resources/views/pages/auth/dummy-dashboard.blade.php]: trail spacing!

If you see someting like this in your PR just add breakline

File: src/resources/views/pages/auth/dummy-verify-wait.blade.php [hunk 21]
```diff
@@ -0,0 +1,13 @@
...
+
+@section('content')
+
+<body>
+    <p>Please Verify your email</p>
+    <p>test( 0 = unverified, 1 = verified) status: {{auth()->user()->status}}</p>
+    <a href='/resend-email'><button>Resend Email Verification</button></a>
+</body>
+
+</html>
```
Comment (hipe01mark) [inline src/resources/views/pages/auth/dummy-verify-wait.blade.php]: trail spacing!

If you see someting like this in your PR just add breakline

File: src/resources/views/pages/auth/login.blade.php [hunk 22]
```diff
@@ -0,0 +1,45 @@
...
+            <svg class="h-32" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1440 320">
+                <path fill="#333333" fill-opacity="1" d="M0,0L60,26.7C120,53,240,107,360,138.7C480,171,600,181,720,160C840,139,960,85,1080,85.3C1200,85,1320,139,1380,165.3L1440,192L1440,320L1380,320C1320,320,1200,320,1080,320C960,320,840,320,720,320C600,320,480,320,360,320C240,320,120,320,60,320L0,320Z"></path>
+                <path fill="#484848" fill-opacity="1" d="M0,128L48,160C96,192,192,256,288,282.7C384,309,480,299,576,272C672,245,768,203,864,170.7C960,139,1056,117,1152,144C1248,171,1344,245,1392,282.7L1440,320L1440,320L1392,320C1344,320,1248,320,1152,320C1056,320,960,320,864,320C768,320,672,320,576,320C480,320,384,320,288,320C192,320,96,320,48,320L0,320Z"></path>
+            </svg>
+        </div>
+    </div>
+
+</body>
+
+</html>
```
Comment (hipe01mark) [inline src/resources/views/pages/auth/login.blade.php]: trail spacing!

If you see something like this in your PR just add breakline

File: src/resources/views/pages/auth/register.blade.php [hunk 23]
```diff
@@ -0,0 +1,154 @@
...
+        <!-- Submit Button -->
+        <button>Submit</button>
+
+        <p>Already have an Account? <a href="/" class="underline">Sign In!</a></p>
+
+    </form>
+    @endauth
+</body>
+
+</html>
```
Comment (hipe01mark) [inline src/resources/views/pages/auth/register.blade.php]: trail spacing!

If you see something like this in your PR just add breakline

File: src/resources/views/pages/layouts/app.blade.php [hunk 24]
```diff
@@ -0,0 +1,11 @@
...
+
+<head>
+    @include('pages.layouts.head')
+</head>
+
+<body>
+    @yield('content')
+</body>
+
+</html>
```
Comment (hipe01mark) [inline src/resources/views/pages/layouts/app.blade.php]: trail spacing!

If you see something like this in your PR just add breakline

File: src/resources/views/pages/layouts/head.blade.php [hunk 25]
```diff
@@ -0,0 +1,4 @@
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    @vite(['resources/css/app.css', 'resources/js/app.js', 'resources/scss/app.scss'])
+    <title>@yield('title')</title>
```
Comment (hipe01mark) [inline src/resources/views/pages/layouts/head.blade.php]: trail spacing!

If you see something like this in your PR just add breakline

File: src/routes/web.php [hunk 26]
```diff
@@ -1,7 +1,24 @@
...
+// Auth Controller
+Route::post('/register', [AuthController::class, 'register']);
+Route::get('/verify-email/{id}/{hash}', [AuthController::class, 'verifyEmail'])->name('verification.verify');
+Route::get('/verify-waiting', [AuthController::class, 'emailVerifyRedirect']);
+Route::get('/logout', [AuthController::class, 'logout']);
+// Resend Email
+Route::get('/resend-email', [AuthController::class, 'resendEmailVerification']);
+
+// Dashboard Controller
+Route::get('/dashboard', [DashboardController::class, 'dashboard']);
```
Comment (hipe01mark) [inline src/routes/web.php]: is this for logged-in users? if so, add auth middleware to it.

File: src/resources/views/pages/auth/login.blade.php (hunk 22 above)
Comment (hipe01mark) [reply src/resources/views/pages/auth/login.blade.php]: image

File: src/resources/views/pages/auth/register.blade.php (hunk 23 above)
Comment (hipe01mark) [reply src/resources/views/pages/auth/register.blade.php]: image

File: src/resources/views/pages/layouts/app.blade.php (hunk 24 above)
Comment (hipe01mark) [reply src/resources/views/pages/layouts/app.blade.php]: image

File: src/resources/views/pages/layouts/head.blade.php (hunk 25 above)
Comment (hipe01mark) [reply src/resources/views/pages/layouts/head.blade.php]: image

File: src/resources/views/pages/auth/dummy-verify-wait.blade.php (hunk 21 above)
Comment (hipe01mark) [reply src/resources/views/pages/auth/dummy-verify-wait.blade.php]: image

File: src/app/Http/Controllers/AuthController.php (hunk 1 above)
Comment (Hakdoff) [reply src/app/Http/Controllers/AuthController.php]: Applied changes in all files.

File: src/resources/views/pages/layouts/head.blade.php [hunk 27]
```diff
@@ -1,4 +1,5 @@
     <meta charset="UTF-8">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
     @vite(['resources/css/app.css', 'resources/js/app.js', 'resources/scss/app.scss'])
-    <title>@yield('title')</title>
\ No newline at end of file
+    <title>@yield('title')</title>
+
```
Comment (hipe01mark) [inline src/resources/views/pages/layouts/head.blade.php]: trail spacing!

**Review by hipe01mark — APPROVED** [review]
one more then GEFN 🙇

**Review by hipe01mark — APPROVED** [review]
GEFN 🙇

**Review by PeterDMadrid — APPROVED** [review]

**Pull request merged**